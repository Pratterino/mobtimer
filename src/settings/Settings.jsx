import React, {Component} from "react";
import {bindActionCreators} from "redux";
import {connect} from "react-redux";
import "./Settings.scss";
import {clearState} from "./../settings/settingsActions";

class Settings extends Component {
    componentDidMount() {
        window.document.body.className = this.props.settings.theme || "";
    }

    render() {
        return (
            <div className="settings">
                <h3>Application</h3>
                <a className="button" onClick={this.props.clearState}>Reset application</a>
            </div>
        );
    }
}

const mapStateToProps = state => ({
    settings: state.settings,
});
const mapDispatchToProps = dispatch => bindActionCreators({
    clearState,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Settings);
