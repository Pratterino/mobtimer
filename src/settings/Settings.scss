$marginHeader: 10px;

.settings__hamburger {
  position: absolute;
  right: 0;
  margin: 20px;
  top: 0;
  z-index: 99999;

  &:hover {
    > div {
      background-color: #333;
    }
  }

  .bar1, .bar2, .bar3 {
    width: 35px;
    height: 5px;
    background-color: var(--header-color);
    margin: 6px 0;
    transition: 0.4s;
  }

  &.change {
    > div {
      background-color: #333;
    }

    .bar1 {
      -webkit-transform: rotate(-45deg) translate(-9px, 6px);
      transform: rotate(-45deg) translate(-9px, 6px);
    }

    .bar2 {
      opacity: 0;
    }

    .bar3 {
      -webkit-transform: rotate(45deg) translate(-8px, -8px);
      transform: rotate(45deg) translate(-8px, -8px);
    }
  }
}

.settings {
  $button_size: 30px;
  z-index: 5;
  overflow-y: auto;
  padding: var(--base-padding);
  background: var(--header-color);
  display: flex;
  position: absolute;
  height: 100vh;
  right: 0;
  top: 0;
  flex-direction: column;
  justify-content: space-evenly;

  list-style: none;

  transform: translateY(-120%);
  transition: transform 400ms ease-in, margin-bottom 400ms ease-in;

  &.slideIn {
    transform: translateX(0);
    box-shadow: var(--background) 0px 0px 0px 5px;
  }

  &.slideOut {
    transform: translateX(120%);
  }

  h3 {
    margin: $marginHeader 0;
    color: #faebd7;
    text-decoration: underline;
    text-decoration-color: var(--link-color-darker);
  }

  .settings__group {
    flex-basis: 33%;
    border: 1px dashed var(--background);

    &:first-of-type {
      border-top-width: 0;
    }

    border-left-width: 0;
    border-right-width: 0;

    .settings__group-subtitle {
      font-size: 10pt;
      color: var(--text-color);
      margin-top: -$marginHeader;
    }
  }

  .left {
    text-align: left;
  }

  a, a:focus, a:visited, button {
    margin: 4px;
    text-decoration: none;
    color: #fff;
    border: 1px solid #faebd7;
    text-align: center;
    letter-spacing: .5px;
    background: transparent;
    transition: background-color .2s ease-out;
    cursor: pointer;
    border-left-width: 2px;
    border-bottom-width: 2px;
    border-radius: 4px;
    display: inline-block;
    height: $button_size;
    line-height: $button_size;
    padding: 0 12px;

    &:active {
      border: none;
    }
  }
}
