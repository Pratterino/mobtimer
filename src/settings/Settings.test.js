import React from 'react';
import {unwrapped as UnwrappedSettings} from './Settings';
import {shallow} from "enzyme";

describe('Settings', () => {
    let wrapper;

    const renderComponent = (props) => (
        wrapper = shallow(
            <UnwrappedSettings
                settings={{
                    devMode: false,
                    theme: "",
                }}
                timer={{
                    sessionLength: 900,
                }}
                {...props}
            />
        )
    );

    beforeEach(() => {
        renderComponent();
    });

    describe('render', () => {
        it('should match snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('should match snapshot when devMode', () => {
            renderComponent({settings: {devMode: true}});
            expect(wrapper).toMatchSnapshot();
        });
    })
});
