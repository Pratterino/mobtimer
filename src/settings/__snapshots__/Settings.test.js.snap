// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings render should match snapshot 1`] = `
<Fragment>
  <div
    className="settings slideOut"
  >
    <div
      className="settings__group"
    >
      <h3>
        Application
      </h3>
      <li>
        <button
          className="button"
        >
          Reset application state
        </button>
      </li>
      <li>
        <a
          href="https://github.com/Pratterino/mobtimer"
          target="__blank"
        >
          https://github.com/Pratterino/mobtimer
        </a>
      </li>
    </div>
    <div
      className="settings__group"
    >
      <h3>
        Timer
      </h3>
      <label
        htmlFor="sessionLength"
      >
        Session length in minutes
      </label>
      <input
        defaultValue={15}
        min="1"
        name="sessionLength"
        onChange={[Function]}
        type="number"
      />
    </div>
    <div
      className="settings__group"
    >
      <h3>
        Theme
      </h3>
      <p
        className="settings__group-subtitle"
      >
        (Current: 
        )
      </p>
      <Connect(ThemeSelector) />
    </div>
    <div
      className="settings__group"
    >
      <h3>
        Help
      </h3>
      <a
        href="https://paypal.me/pratterino"
        target="__blank"
      >
        Donate a beer 
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f4b9",
                "M256 416c114.9 0 208-93.1 208-208S370.9 0 256 0 48 93.1 48 208s93.1 208 208 208zM233.8 97.4V80.6c0-9.2 7.4-16.6 16.6-16.6h11.1c9.2 0 16.6 7.4 16.6 16.6v17c15.5.8 30.5 6.1 43 15.4 5.6 4.1 6.2 12.3 1.2 17.1L306 145.6c-3.8 3.7-9.5 3.8-14 1-5.4-3.4-11.4-5.1-17.8-5.1h-38.9c-9 0-16.3 8.2-16.3 18.3 0 8.2 5 15.5 12.1 17.6l62.3 18.7c25.7 7.7 43.7 32.4 43.7 60.1 0 34-26.4 61.5-59.1 62.4v16.8c0 9.2-7.4 16.6-16.6 16.6h-11.1c-9.2 0-16.6-7.4-16.6-16.6v-17c-15.5-.8-30.5-6.1-43-15.4-5.6-4.1-6.2-12.3-1.2-17.1l16.3-15.5c3.8-3.7 9.5-3.8 14-1 5.4 3.4 11.4 5.1 17.8 5.1h38.9c9 0 16.3-8.2 16.3-18.3 0-8.2-5-15.5-12.1-17.6l-62.3-18.7c-25.7-7.7-43.7-32.4-43.7-60.1.1-34 26.4-61.5 59.1-62.4zM480 352h-32.5c-19.6 26-44.6 47.7-73 64h63.8c5.3 0 9.6 3.6 9.6 8v16c0 4.4-4.3 8-9.6 8H73.6c-5.3 0-9.6-3.6-9.6-8v-16c0-4.4 4.3-8 9.6-8h63.8c-28.4-16.3-53.3-38-73-64H32c-17.7 0-32 14.3-32 32v96c0 17.7 14.3 32 32 32h448c17.7 0 32-14.3 32-32v-96c0-17.7-14.3-32-32-32z",
              ],
              "iconName": "donate",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
         =&gt; 
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f0fc",
                "M368 96h-48V56c0-13.255-10.745-24-24-24H24C10.745 32 0 42.745 0 56v400c0 13.255 10.745 24 24 24h272c13.255 0 24-10.745 24-24v-42.11l80.606-35.977C429.396 365.063 448 336.388 448 304.86V176c0-44.112-35.888-80-80-80zm16 208.86a16.018 16.018 0 0 1-9.479 14.611L320 343.805V160h48c8.822 0 16 7.178 16 16v128.86zM208 384c-8.836 0-16-7.164-16-16V144c0-8.836 7.164-16 16-16s16 7.164 16 16v224c0 8.836-7.164 16-16 16zm-96 0c-8.836 0-16-7.164-16-16V144c0-8.836 7.164-16 16-16s16 7.164 16 16v224c0 8.836-7.164 16-16 16z",
              ],
              "iconName": "beer",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </a>
      <p>
        If you enjoy this mob timer and want to support it.
        <br />
        A small donation would be really really appreciated!
        <br />
      </p>
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f004",
              "M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z",
            ],
            "iconName": "heart",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
       
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f004",
              "M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z",
            ],
            "iconName": "heart",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
       
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f004",
              "M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z",
            ],
            "iconName": "heart",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </div>
  </div>
  <div
    className="settings__hamburger pointer "
    onClick={[Function]}
  >
    <div
      className="bar1"
    />
    <div
      className="bar2"
    />
    <div
      className="bar3"
    />
  </div>
</Fragment>
`;

exports[`Settings render should match snapshot when devMode 1`] = `
<Fragment>
  <div
    className="settings slideOut"
  >
    <div
      className="settings__group"
    >
      <h3>
        Application
      </h3>
      <li>
        <button
          className="button"
        >
          Reset application state
        </button>
      </li>
      <li>
        <a
          href="https://github.com/Pratterino/mobtimer"
          target="__blank"
        >
          https://github.com/Pratterino/mobtimer
        </a>
      </li>
    </div>
    <div
      className="settings__group"
    >
      <h3>
        Timer
      </h3>
      <label
        htmlFor="sessionLength"
      >
        Session length in minutes
      </label>
      <input
        defaultValue={15}
        min="1"
        name="sessionLength"
        onChange={[Function]}
        type="number"
      />
    </div>
    <div
      className="settings__group"
    >
      <h3>
        Theme
      </h3>
      <p
        className="settings__group-subtitle"
      >
        (Current: 
        )
      </p>
      <Connect(ThemeSelector) />
    </div>
    <div
      className="settings__group"
    >
      <h3>
        Help
      </h3>
      <a
        href="https://paypal.me/pratterino"
        target="__blank"
      >
        Donate a beer 
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f4b9",
                "M256 416c114.9 0 208-93.1 208-208S370.9 0 256 0 48 93.1 48 208s93.1 208 208 208zM233.8 97.4V80.6c0-9.2 7.4-16.6 16.6-16.6h11.1c9.2 0 16.6 7.4 16.6 16.6v17c15.5.8 30.5 6.1 43 15.4 5.6 4.1 6.2 12.3 1.2 17.1L306 145.6c-3.8 3.7-9.5 3.8-14 1-5.4-3.4-11.4-5.1-17.8-5.1h-38.9c-9 0-16.3 8.2-16.3 18.3 0 8.2 5 15.5 12.1 17.6l62.3 18.7c25.7 7.7 43.7 32.4 43.7 60.1 0 34-26.4 61.5-59.1 62.4v16.8c0 9.2-7.4 16.6-16.6 16.6h-11.1c-9.2 0-16.6-7.4-16.6-16.6v-17c-15.5-.8-30.5-6.1-43-15.4-5.6-4.1-6.2-12.3-1.2-17.1l16.3-15.5c3.8-3.7 9.5-3.8 14-1 5.4 3.4 11.4 5.1 17.8 5.1h38.9c9 0 16.3-8.2 16.3-18.3 0-8.2-5-15.5-12.1-17.6l-62.3-18.7c-25.7-7.7-43.7-32.4-43.7-60.1.1-34 26.4-61.5 59.1-62.4zM480 352h-32.5c-19.6 26-44.6 47.7-73 64h63.8c5.3 0 9.6 3.6 9.6 8v16c0 4.4-4.3 8-9.6 8H73.6c-5.3 0-9.6-3.6-9.6-8v-16c0-4.4 4.3-8 9.6-8h63.8c-28.4-16.3-53.3-38-73-64H32c-17.7 0-32 14.3-32 32v96c0 17.7 14.3 32 32 32h448c17.7 0 32-14.3 32-32v-96c0-17.7-14.3-32-32-32z",
              ],
              "iconName": "donate",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
         =&gt; 
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f0fc",
                "M368 96h-48V56c0-13.255-10.745-24-24-24H24C10.745 32 0 42.745 0 56v400c0 13.255 10.745 24 24 24h272c13.255 0 24-10.745 24-24v-42.11l80.606-35.977C429.396 365.063 448 336.388 448 304.86V176c0-44.112-35.888-80-80-80zm16 208.86a16.018 16.018 0 0 1-9.479 14.611L320 343.805V160h48c8.822 0 16 7.178 16 16v128.86zM208 384c-8.836 0-16-7.164-16-16V144c0-8.836 7.164-16 16-16s16 7.164 16 16v224c0 8.836-7.164 16-16 16zm-96 0c-8.836 0-16-7.164-16-16V144c0-8.836 7.164-16 16-16s16 7.164 16 16v224c0 8.836-7.164 16-16 16z",
              ],
              "iconName": "beer",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </a>
      <p>
        If you enjoy this mob timer and want to support it.
        <br />
        A small donation would be really really appreciated!
        <br />
      </p>
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f004",
              "M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z",
            ],
            "iconName": "heart",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
       
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f004",
              "M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z",
            ],
            "iconName": "heart",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
       
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f004",
              "M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z",
            ],
            "iconName": "heart",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </div>
  </div>
  <div
    className="settings__hamburger pointer "
    onClick={[Function]}
  >
    <div
      className="bar1"
    />
    <div
      className="bar2"
    />
    <div
      className="bar3"
    />
  </div>
</Fragment>
`;
